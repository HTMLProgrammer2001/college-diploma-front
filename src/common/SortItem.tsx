import React from 'react';
import cn from 'classnames';


type ISortItemProps = {
	state: -1 | 1 | undefined,
	change: (param: string) => void,
	param: string
};

const SortItem: React.FC<ISortItemProps> = ({state, change, param}) => (
	<i className={cn("fa cur pull-right", {
		["fa-sort-amount-asc"]: state == 1 || state == undefined,
		["fa-sort-amount-desc"]: state == -1,
		["opacity-5"]: state == undefined,
	})}
	   onClick={() => change(param)}
	/>
);

export default SortItem;
