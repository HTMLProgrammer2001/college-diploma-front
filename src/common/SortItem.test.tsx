import React from 'react';
import {shallow} from 'enzyme';

import SortItem from './SortItem';


describe('Test sort item element', () => {
	it('Test render', () => {
		const wrapper = shallow(<SortItem state={1} change={() => null} param={''}/>);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Test click', () => {
		const clickHandler = jest.fn(),
			  param = 'name',
			  wrapper = shallow(<SortItem state={1} change={clickHandler} param={param}/>);

		//click
		wrapper.simulate('click');

		//tests
		expect(clickHandler).toBeCalledTimes(1);
		expect(clickHandler).toBeCalledWith(param);
	});

	it('Test state changes', () => {
		const wrapper = shallow(<SortItem state={1} change={() => null} param={''}/>);
		expect(wrapper.find('.fa-sort-amount-asc')).toBeTruthy();

		wrapper.setProps({state: -1});
		expect(wrapper.find('.fa-sort.amount-desc')).toBeTruthy();

		wrapper.setProps({state: undefined});
		expect(wrapper.find('.opacity-5')).toBeTruthy();
	});
});
