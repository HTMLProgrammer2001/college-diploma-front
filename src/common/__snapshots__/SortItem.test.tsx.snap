// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sort item element Test render 1`] = `"<i class=\\"fa cur pull-right fa-sort-amount-asc\\"></i>"`;
